### PR TITLE
Adding telemetry publish command to task lib

### PR DESCRIPTION
**Pull request by nigurr**

### CODE REVIEW

**Comment by nigurr** [comment]
@ericsciple, @TingluoHuang  can you please review and approve the same?

File: node/task.ts [hunk 1]
```diff
@@ -1645,3 +1645,26 @@ if (semver.lt(process.versions.node, '4.2.0')) {
...
+ */
+export function publishTelemetry(area: string, feature: string, properties: { [key: string]: any }): void {
+
+    let agent = getVariable('Agent.Version');
+    if (!agent || semver.lt(agent, '2.120.0')) {
+        return;
+    }
+
+    let data = JSON.stringify(properties);
+    debug('telemetry area: ' + area + ' feature: ' + feature + ' data: ' + data);
```
Comment (ericsciple) [inline node/task.ts]: @TingluoHuang to confirm, i think the pattern we normally use is the agent prints the debug info related to the `##` commands.

Comment (TingluoHuang) [inline node/task.ts]: yes, agent will print out any ## commands it run when system.debug=true

File: node/test/telemetrytests.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    it('publish telemetry if the agent version is greater than 2.120', function (done) {
+        this.timeout(1000);
+
+        let originalAgentVersion = process.env['AGENT_VERSION'];
+        process.env['AGENT_VERSION'] = '2.120.0';
+
+        try {
+            let data = { 'abc': 'xyz' };
+            var stdStream = testutil.createStringStream();
```
Comment (ericsciple) [inline node/test/telemetrytests.ts]: does this need to unset stdstream in the finally?

File: node/test/telemetrytests.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/// <reference path="../typings/index.d.ts" />
+/// <reference path="../_build/task.d.ts" />
+
+import assert = require('assert');
+import * as tl from '../_build/task';
+import testutil = require('./testutil');
+
+describe('Telemetry Tests', function () {
```
Comment (ericsciple) [inline node/test/telemetrytests.ts]: can you move this into commandtests.ts to keep the number of files down (there are only two tests in this file)

**Comment by ericsciple** [comment]
@TingluoHuang to review. afaik i know this makes sense but i believe ting was involved more on the `##` command on the agent side.

File: node/test/telemetrytests.ts (hunk 3 above)
Comment (nigurr) [inline node/test/telemetrytests.ts]: commandtest tests are targeted at 'taskcommand' file rather than specific command. so i kept separate file.

**Comment by nigurr** [comment]
@TingluoHuang Can you please complete the review/PR?

**Pull request merged**